module/zfs/zvol.c

```diff
@@ -466,8 +473,15 @@ zvol_set_volsize(const char *name, uint64_t volsize)
 	if (zv != NULL)
 		mutex_exit(&zv->zv_state_lock);
 
-	if (disk != NULL)
+	if (disk != NULL) {
+#ifdef HAVE_REVALIDATE_DISK_SIZE
+		int ret = 0;
+		ret = zvol_revalidate_disk(disk);
+		revalidate_disk_size(disk, ret == 0);
```

Comment:
You can avoid the need for the forward reference like so.

```suggestion
		revalidate_disk_size(disk, disk->fops->revalidate_disk(disk) == 0);
```

Looking at the master version of this commit 59b68723  I see that we're actually not calling `zvol_revalidate_disk()` for 5.10 kernels.  Would you mind opening another PR against the master branch to make this same fix to `zvol_update_volsize()`.

Reply:
I'll push out another commit that removes the forward declaration.

Reply:
> You can avoid the need for the forward reference like so.
> 
> Looking at the master version of this commit [59b6872](https://github.com/openzfs/zfs/commit/59b687232798a4da0e7647cca19a308aa82b5fad) I see that we're actually not calling `zvol_revalidate_disk()` for 5.10 kernels. Would you mind opening another PR against the master branch to make this same fix to `zvol_update_volsize()`.

Sure thing.

Reply:
Be advised there's one more significant commit #11351 required for 5.10 which hasn't yet been merged to master.  It won't cause a build failure but it will result in `sendfile(2)` not working.  It's also probably going to be a bit of a pain to backport.

Reply:
> Be advised there's one more significant commit #11351 required for 5.10 which hasn't yet been merged to master. It won't cause a build failure but it will result in `sendfile(2)` not working. It's also probably going to be a bit of a pain to backport.

Yeah, I saw that pull request jut after I submitted mine (and cringed).  I'll take a stab at backporting it once you've got it merged to master.  Until then, I'll tentatively test with 5.10 and hope nothing uses `sendfile(2)`.